cloudify_agent/api/utils.py

```diff
@@ -510,3 +513,55 @@ def json_loads(content):
         return json.loads(content)
     except ValueError as e:
         raise ValueError('{0}:{1}{2}'.format(str(e), os.linesep, content))
+
+
+def safe_create_dir(path):
+    # creating a dir, ignoring exists error to handle possible race condition
+    try:
+        os.makedirs(path)
+    except OSError as ose:
+        if ose.errno != errno.EEXIST:
+            raise
+
+
+def get_rest_client(security_enabled,
+                    rest_host,
+                    rest_protocol,
+                    rest_port,
+                    rest_username=None,
+                    rest_password=None,
+                    verify_rest_certificate=None,
+                    ssl_cert_path=None):
+
+    if not security_enabled:
+        rest_client = CloudifyClient(host=rest_host,
+                                     protocol=rest_protocol,
+                                     port=rest_port)
+    else:
+        if not rest_username or not rest_password:
+            raise ValueError('username or password are missing! Both are '
+                             'required to create a REST client for a secured '
+                             'manager [{0}]'.format(rest_host))
+
+        credentials = '{0}:{1}'.format(rest_username, rest_password)
+        headers = {
+            'Authorization': 'Basic ' + base64_encode(credentials)}
+
+        if verify_rest_certificate.lower() == 'true':
```

Comment:
verify_rest_certificate should be made into a boolean. moreover, how is security_enabled a boolean while this isn't?